### PR TITLE
chore(collector): upgrade to app version 0.37.1

### LINKED ISSUE #42: Regex for extract_metadata_from_filepath added by #36 is too strict
The regex is currently:
`^.*\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[a-f0-9\-]{36})\/(?P<container_name>[^\._]+)\/(?P<run_id>\d+)\.log$`

The `uid` check for 36 characters is too strict. I had to change the capture group to `(?P<uid>[^\/]+)`, which also matches with what is currently in PR #38

### LINKED ISSUE #88: v0.36.0 drop support for mem-ballast-size-mib
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/21709e3054dbfb876d40114469233971e2c88bac/charts/opentelemetry-collector/templates/_pod.tpl#L15

should be dropped to support latest v0.36.0 tag of the collector image.

see this https://github.com/open-telemetry/opentelemetry-collector/pull/3626

### PR DESCRIPTION
**Pull request by secustor**
As #80  seems to be abandoned I have created this new PR to accommodate the changes in the meantime. 

appVersion is now prefixed with `v` as the tagging has changed with version 0.36.0 

@dmitryax 

Fixes #88
Fixes #42

### CODE REVIEW

**Comment by bbeattie-phxlabs** [comment]
Apologies. Was blocked in going through the necessary hoops to get the CLA signed. That said, happy to defer to your merge that does similarly. Note that upgrading to 0.36.0 additionally requires some configuration syntax changes (see https://github.com/open-telemetry/opentelemetry-helm-charts/pull/80/files#diff-d3c8687b50b2f7b2ca10ff878367b16b76ead0cfdf62548091b5bcc507dc2d68L114).

**Comment by bbeattie-phxlabs** [comment]
No worries. Might also be worth looking into the feedback provided in the other pull request, e.g. https://github.com/open-telemetry/opentelemetry-helm-charts/pull/80#issuecomment-926376133

File: charts/opentelemetry-collector/Chart.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ maintainers:
   - name: pjanotti
   - name: tigrannajaryan
 icon: https://opentelemetry.io/img/logos/opentelemetry-logo-nav.png
-appVersion: 0.22.0
+appVersion: v0.36.0
```
Comment (dmitryax) [inline charts/opentelemetry-collector/Chart.yaml:16]: The default configuration has jaeger and zipkin receivers which are not part of the core image anymore. So we should switch default image to contrib as well

File: charts/opentelemetry-collector/values.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ config:
     logging: {}
   extensions:
     health_check: {}
+    memory_ballast: {}
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.yaml:13]: In order to take effect, this also needs to be added to service.extensions

**Pull request merged**